modules/importer/src/runner/progress.rs

```diff
@@ -0,0 +1,189 @@
+use std::{
+    future::Future,
+    time::{Duration, Instant},
+};
+use tokio::runtime::Handle;
+
+pub trait Progress {
+    type Instance: ProgressInstance;
+
+    fn start(&self, work: usize) -> Self::Instance;
+}
+
+impl Progress for () {
+    type Instance = ();
+
+    fn start(&self, _work: usize) -> Self::Instance {}
+}
+
+impl ProgressInstance for () {
+    async fn increment(&mut self, _work: usize) {}
+
+    async fn finish(self) {}
+}
+
+pub trait ProgressInstance: Sized {
```

Comment:
Would it be worthwhile (or possible?) to know some total-expected-ticks?  Able to give us not just "things are progressing" but "you're 85% done" sort thing?

Reply:
That's what it should do today. For example:

```
2024-08-09T14:01:51.798535Z  INFO run_sync:walk: run: Import OSV: https://github.com/RConsortium/r-advisory-database: 10.00% (ETA: 0s) percent=10.0 base="/tmp/.tmpivFAHw/vulns"
```

ETA 0s, because it's rounded down to seconds, and it's below that in this case.

Reply:
Ok, added `total` and `current` to the trace message too:

```
2024-08-09T14:30:52.929783Z  INFO run_sync:walk: run: Import OSV: https://github.com/RConsortium/r-advisory-database: 10.00% (ETA: 0s) current=1 total=10 percent=10.0 base="/tmp/.tmpTWGs40/vulns"
```